### PR TITLE
Redirect bug

### PR DESCRIPTION
**Pull request by bdc34**
A GET to /login should redirect to /user when the client is already logged in. This works but it ignores any next_page query param the client sends.

I've included a fix and a unit test for this in this PR. 

The unit test doesn't work in spite of being almost identical to other unit tests in the suite. Send me any advice or ideas you might have about this. I'm running the unit test with

`WITH_INTEGRATION=1 pipenv run pytest 
    --cov=accounts 
    --cov=users/arxiv 
    --cov-report=term-missing 
    accounts users/arxiv`

### CODE REVIEW

File: accounts/accounts/tests/test_end_to_end.py [hunk 1]
```diff
@@ -415,6 +419,22 @@ def test_post_login(self):
...
+
+    def test_already_logged_in_redirect(self):
+        """User logs in, then reqeusts /login again and should be redirected"""
+        client = self.app.test_client()
+        client.environ_base = self.environ_base
+        form_data = {'username': 'foouser', 'password': 'thepassword'}
+        response = client.post('/login', data=form_data)
+        self.assertEqual(response.status_code, status.HTTP_303_SEE_OTHER)
+
+        next_page = 'https://arxiv.org/some_sort_of_next_page?cheeseburger=yes%20please'
```
Comment (bmaltzan) [inline accounts/accounts/tests/test_end_to_end.py:431]: Is the domain ok to include in next_page, in the client.get call?

**Comment by bmaltzan** [comment]
These args reduce the output a little:
pytest -k test_get_login  accounts/accounts/tests/test_end_to_end.py

File: accounts/accounts/tests/test_end_to_end.py (hunk 1 above)
Comment (bmaltzan) [reply accounts/accounts/tests/test_end_to_end.py:431]: In routes/ui.py#anonymous_only#wrapper
request.auth is None, when called by the test's client.get.
Maybe it needs to be adjusted to send cookies?

Comment (bdc34) [reply accounts/accounts/tests/test_end_to_end.py:431]: > 

It does seem like a danger. 
It is critical to be able to go to URLs like https://mod.arxiv.org/something or services.dev.arxiv.org. 
We could restrict it to just subdomains of arxiv.org or have a regex.

Comment (bdc34) [reply accounts/accounts/tests/test_end_to_end.py:431]: Re: request.auth is None, when called by the test's client.get. 

I checked that the cookies are sent and they seem to be there on the server. reqeust.auth gets set by code in arxiv-auth/user and I'm wondering if that is not setup in the tests.

Comment (bdc34) [reply accounts/accounts/tests/test_end_to_end.py:486]: @bmaltzan Thanks again, you advice was helpful.

**Comment by bdc34** [comment]
I've added an end to end test that logs in and actually tries to get a protected endpoint. 

I looked through all the tests in arxiv-auth and didn't find this kind of test. It seems critical so I added it.

It is test_login_get_something_protected() in accounts/accounts/tests/test_end_to_end.py

**Comment by bdc34** [comment]
The test pass on my laptop but this is failing in travis with: https://app.travis-ci.com/github/arXiv/arxiv-auth/builds/243063631

**Pull request merged**